src/charm.py

```diff
@@ -170,6 +191,48 @@ def _configure_charm_authorization(self):
             login_details_secret = self.model.get_secret(label=GOCERT_LOGIN_SECRET_LABEL)
             login_details_secret.set_content(login_details.to_dict())
 
+    def _configure_certificate_requirers(self):
+        """Get all CSR's and certs from databags and Gocert, compare differences and update requirers if needed."""
+        login_details = self._get_or_create_admin_account()
+        if not login_details or not login_details.token:
+            logger.warning("couldn't distribute certificates: not logged in")
+            return
+        databag_csrs = self.tls.get_certificate_requests()
+        gocert_csrs_table = self.client.get_certificate_requests_table(login_details.token)
+        if not gocert_csrs_table:
```

Comment:
The csrs come from the databags and are posted to gocert. If the user uploads a certificate to gocert, that certificate is then pushed to the databag. We can't push the CSRs from the table to relations because we wouldn't know which relation they would be posted to.

> we post the csrs here if the table is empty and the databg has csrs
That is exactly what happens.

Reply:
I see, thanks for the clarification.